crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -491,22 +569,171 @@ impl<'a> SimpleTokenizer<'a> {
+                } else {
+                    SimpleTokenKind::Ampersand
+                }
+            }
+            '-' => {
+                if self.cursor.eat_char('=') {
+                    SimpleTokenKind::MinusEqual
+                } else if self.cursor.eat_char('>') {
+                    SimpleTokenKind::RArrow
+                } else {
+                    SimpleTokenKind::Minus
+                }
+            }
+            '@' => {
+                if self.cursor.eat_char('=') {
+                    SimpleTokenKind::AtEqual
+                } else {
+                    SimpleTokenKind::At
+                }
+            }
+            '!' => {
+                if self.cursor.eat_char('=') {
+                    SimpleTokenKind::NotEqual
+                } else {
+                    self.bogus = true;
+                    SimpleTokenKind::Other
+                }
+            }
+            '~' => SimpleTokenKind::Tilde,
+            ':' => {
+                if self.cursor.eat_char('=') {
+                    SimpleTokenKind::ColonEqual
+                } else {
+                    SimpleTokenKind::Colon
+                }
+            }
+            ';' => SimpleTokenKind::Semi,
+            '<' => {
+                if self.cursor.eat_char('<') {
+                    if self.cursor.eat_char('=') {
+                        SimpleTokenKind::LeftShiftEqual
+                    } else {
+                        SimpleTokenKind::LeftShift
+                    }
+                } else if self.cursor.eat_char('=') {
+                    SimpleTokenKind::LessEqual
+                } else {
+                    SimpleTokenKind::Less
+                }
+            }
+            '>' => {
+                if self.cursor.eat_char('>') {
+                    if self.cursor.eat_char('=') {
+                        SimpleTokenKind::RightShiftEqual
+                    } else {
+                        SimpleTokenKind::RightShift
+                    }
+                } else if self.cursor.eat_char('=') {
+                    SimpleTokenKind::GreaterEqual
+                } else {
+                    SimpleTokenKind::Greater
+                }
+            }
+            ',' => SimpleTokenKind::Comma,
+            '.' => {
+                if self.cursor.first() == '.' && self.cursor.second() == '.' {
+                    self.cursor.bump();
+                    self.cursor.bump();
+                    SimpleTokenKind::Ellipsis
+                } else {
+                    SimpleTokenKind::Dot
+                }
+            }
+
+            // Bracket tokens
+            '(' => SimpleTokenKind::LParen,
+            ')' => SimpleTokenKind::RParen,
+            '[' => SimpleTokenKind::LBracket,
+            ']' => SimpleTokenKind::RBracket,
+            '{' => SimpleTokenKind::LBrace,
+            '}' => SimpleTokenKind::RBrace,
+
+            // Keyword tokens
+            c if is_identifier_start(c) => {
```

Comment:
Identifiers are the most common tokens. It can make sense to have an extra branch at the top that handles identifiers (similar to lexer)

```
c if is_ascii_identifier_start(c) => self.lex_identifier(c)?,
```